src/Services/OrderService/OrderServiceWebAPI/Endpoints/CreateOrder.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace OrderServiceWebAPI.Endpoints;
+
+public record CreateOrderResponse(Guid Id);
+
+public class CreateOrder : ICarterModule
+{
+    public void AddRoutes(IEndpointRouteBuilder app)
+    {
+        app.MapPost("/orders", async (IPublishEndpoint publishEndpoint, CreateOrderCommand command, ISender sender) =>
+        {
+            var request = command.Adapt<CreateOrderCommand>();
+
+            var result = await sender.Send(request);
+
+            var response = result.Adapt<CreateOrderResponse>();
+
+            var orderId = response.Id;
+
+            var orderProcessedEvent = new OrderProcessedIntegrationEvent(
+                OrderId: orderId,
+                CustomerId: command.CustomerId,
+                ShippingAddress: new Address(
+                    FirstName: command.ShippingAddress.FirstName,
+                    LastName: command.ShippingAddress.LastName,
+                    EmailAddress: command.ShippingAddress.EmailAddress,
+                    Country: command.ShippingAddress.Country,
+                    State: command.ShippingAddress.State,
+                    Street: command.ShippingAddress.Street
+                ),
+                Status: command.Status.ToIntegrationEnum(),
+                Items: command.Items.Select(i => new OrderItem(
+                    BookId: i.BookId,
+                    Title: i.Title,
+                    SellerId: i.SellerId,
+                    Quantity: i.Quantity,
+                    Price: i.Price
+                )).ToList(),
+                TotalPrice: command.Items.Sum(i => i.Quantity * i.Price)
+            );
+
+            await publishEndpoint.Publish(orderProcessedEvent);
+
+            return Results.Created($"/orders/{response.Id}", response);
+        })
+        .WithName("CreateOrder")
+        .Produces<CreateOrderResponse>(StatusCodes.Status201Created)
+        .ProducesProblem(StatusCodes.Status400BadRequest)
+        .WithSummary("Create Order")
+        .WithDescription("Create Order");
+    }
+}
+
+public static class DomainIntegrationEnumsConverter
```

Comment:
move to separate file.
is not it similar to application enum mapper? would be good if you move it there is they are different or if they are same then you that class from application